docker/rucio_client/scripts/setSiteAvailability

```diff
@@ -62,17 +63,34 @@ for site in sites:
             continue
         rucio_status = site.get('rucio_status', None)
         if not rucio_status or rucio_status not in ['dependable', 'enabled']:
-            available_map[rse] = False
+            available_map[rse] = {
+                "availability_read": False,
+                "availability_write": False,
+                "availability_delete": False
+            }
         else:
-            available_map[rse] = True
+            available_map[rse] = {
+                "availability_read": True,
+                "availability_write": True,
+                "availability_delete": True
+            }
 
-print("Skipping RSEs: %s" % skip_rses)
+        # If the site is enabled based on SSB, do an occupancy check
+        if available_map[rse]:
+            should_enable = should_enable_availability_on_occupancy(rse)
+            if not should_enable:
+                print(f"The following site doesn't have enough free space. Availability write should be disabled: {rse}")
+                available_map[rse]["availability_write"] = False
 
-for rse, available in available_map.items():
+
+print("Following RSE are skipped due to skip_site_availability_update: %s" % skip_rses)
+
+for rse, decision in available_map.items():
     try:
         if not DRY_RUN:
-            rclient.update_rse(rse, {"availability_write": available, "availability_delete": available, "availability_read": available})
-        print('Setting availability for %s to %s' % (rse, available))
+            rclient.update_rse(rse, {"availability_write": decision["availability_write"], "availability_delete": decision["availability_delete"], "availability_read": decision["availability_read"]})
```

Comment:
This line is probably too long. Break it up nicely.

Reply:
Done